### ISSUE TITLE
The type of  the `transform` option of `useQuery` does not allow with type change

### ISSUE THREAD

**Issue by mlbiche** [wobsoriano/trpc-nuxt#92]
Due to a type constraint the `transform` option can not be used to transform data to a different type. For example :

```ts
$client.todo.getTodos.useQuery(undefined, {
  transform: ({id, description}) => new Map(id, description)
})
```

is not allowed by the type constraint. The generic type parameters of `useQuery` also do not allow providing a transform output type as Nuxt `useAsyncData` has :

```ts
function useAsyncData<ResT, DataE = Error, DataT = ResT, PickKeys extends KeysOf<DataT>
```

Providing `DataT` to `useQuery` would be great to define a transform return type. We could do something like this then :

```ts
import { TRPCClientError } from '@trpc/client'
import type { inferRouterOutputs } from '@trpc/server'
import type { AppRouter } from '@/server/trpc/routers'

type RouterOutput = inferRouterOutputs<AppRouter>
type GetTodosOutput = RouterOutput['todo']['getTodos']

type ErrorOutput = TRPCClientError<AppRouter>

const { $client } = useNuxtApp()
return $client.todo.getTodos.useQuery<GetTodosOutput, ErrorOutput, Map<string, string>>(undefined, {
  transform: ({id, description}) => new Map(id, description)
})
```

**Comment by wobsoriano** [comment]
Fixed in c32fdf78939e617cc180ef9e7d1b18573205b881

**Comment by mlbiche** [comment]
Wahouh, you rocks ! Cheers !! 🎉
